tests/integration/models_containers_test.py

```diff
@@ -279,6 +279,28 @@ def test_stats(self):
                     'memory_stats', 'blkio_stats']:
             assert key in stats
 
+    def test_ports(self):
+        client = docker.from_env(version=TEST_API_VERSION)
+        target_ports = {'2222/tcp': None}
+        container = client.containers.run(
+            "alpine", "sleep 100", detach=True,
+            ports=target_ports
+        )
+        self.tmp_containers.append(container.id)
+        container.reload()  # required to get auto-assigned ports
+        actual_ports = container.ports
+        assert sorted(target_ports.keys()) == sorted(actual_ports.keys())
+        for target_client, target_host in target_ports.items():
+            for actual_port in actual_ports[target_client]:
+                actual_keys = sorted(actual_port.keys())
+                assert sorted(['HostIp', 'HostPort']) == actual_keys
+                if target_host is None:
+                    int(actual_port['HostPort'])
```

Comment:
shouldn't there be an assert here?

Reply:
No, the `int()` will already throw an error with a reasonably good message if it is not an integer, and that will show up nicely as a failure. And the actual port returned (on the host) is going to be effectively random in this case.

Reply:
makes sense